### PR TITLE
Allow to unset rate limiting contract param

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

We weren't allowing the unset of the rate limiting contract param by setting the contract address to `""`

## Brief Changelog

An empty string is now a valid contract address for rate limits

## Testing and Verifying

Added a test for this.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs [hunk 1]
```diff
@@ -471,4 +471,12 @@ pub mod tests {
             _ => panic!("parsed into wrong variant"),
         }
     }
+
+    #[test]
+    fn packet_with_memo() {
+        // extra fields (like memo) get ignored.
+        let json = r#"{"recv_packet":{"packet":{"sequence":1,"source_port":"transfer","source_channel":"channel-0","destination_port":"transfer","destination_channel":"channel-0","data":{"denom":"stake","amount":"1","sender":"osmo177uaalkhra6wth6hc9hu79f72eq903kwcusx4r","receiver":"osmo1fj6yt4pwfea4865z763fvhwktlpe020ef93dlq","memo":"some info"},"timeout_height":{"revision_height":100}}}}"#;
+        let _parsed: SudoMsg = serde_json_wasm::from_str(json).unwrap();
+        //println!("{parsed:?}");
```
Comment (p0mvn) [inline x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs:480]:
```suggestion
```

File: x/ibc-rate-limit/ibc_middleware_test.go [hunk 2]
```diff
@@ -493,3 +493,18 @@ func (suite *MiddlewareTestSuite) TestFailedSendTransfer() {
...
+	addr := suite.chainA.InstantiateRLContract(&suite.Suite, "")
+	suite.chainA.RegisterRateLimitingContract(addr)
+
+	// Unset the contract param
+	params, err := types.NewParams("")
+	suite.Require().NoError(err)
+	osmosisApp := suite.chainA.GetOsmosisApp()
+	paramSpace, ok := osmosisApp.AppKeepers.ParamsKeeper.GetSubspace(types.ModuleName)
+	suite.Require().True(ok)
+	paramSpace.SetParamSet(suite.chainA.GetContext(), &params)
```
Comment (p0mvn) [inline x/ibc-rate-limit/ibc_middleware_test.go:509]:
```suggestion
        // N.B.: this panics if validation fails.
	paramSpace.SetParamSet(suite.chainA.GetContext(), &params)
```

**Review by p0mvn — APPROVED** [review]

File: x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs [hunk 3]
```diff
@@ -471,4 +471,12 @@ pub mod tests {
             _ => panic!("parsed into wrong variant"),
         }
     }
+
+    #[test]
+    fn packet_with_memo() {
+        // extra fields (like memo) get ignored.
+        let json = r#"{"recv_packet":{"packet":{"sequence":1,"source_port":"transfer","source_channel":"channel-0","destination_port":"transfer","destination_channel":"channel-0","data":{"denom":"stake","amount":"1","sender":"osmo177uaalkhra6wth6hc9hu79f72eq903kwcusx4r","receiver":"osmo1fj6yt4pwfea4865z763fvhwktlpe020ef93dlq","memo":"some info"},"timeout_height":{"revision_height":100}}}}"#;
+        let _parsed: SudoMsg = serde_json_wasm::from_str(json).unwrap();
```
Comment (czarcas7ic) [inline x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs:479]: Glad we tested this!!

**Review by czarcas7ic — APPROVED** [review]
LGTM!

**Review by ValarDragon — APPROVED** [review]
Nice!

**Pull request merged**